plugins/system_upgrade.py

```diff
@@ -62,16 +62,20 @@
 CANT_RESET_RELEASEVER = _(
     "Sorry, you need to use 'download --releasever' instead of '--network'")
 
-STATE_VERSION = 2
+STATE_VERSION = 3
 
 # --- Miscellaneous helper functions ------------------------------------------
 
 
-def reboot():
+def reboot(poweroff_after = None):
     if os.getenv("DNF_SYSTEM_UPGRADE_NO_REBOOT", default=False):
         logger.info(_("Reboot turned off, not rebooting."))
     else:
-        Popen(["systemctl", "reboot"])
+        if poweroff_after is True:
```

Comment:
After the above comments are taken into account this could be simplified to `if poweroff:`